aom/src/main/java/com/nedap/archie/rminfo/MetaModel.java

```diff
@@ -0,0 +1,314 @@
+package com.nedap.archie.rminfo;
+
+import com.nedap.archie.aom.CPrimitiveObject;
+import com.nedap.archie.aom.primitives.CInteger;
+import com.nedap.archie.aom.primitives.CString;
+import com.nedap.archie.aom.profile.AomProfile;
+import com.nedap.archie.aom.profile.AomTypeMapping;
+import com.nedap.archie.base.MultiplicityInterval;
+import com.nedap.archie.paths.PathSegment;
+import com.nedap.archie.query.APathQuery;
+import org.openehr.bmm.core.BmmClass;
+import org.openehr.bmm.core.BmmContainerProperty;
+import org.openehr.bmm.core.BmmEnumeration;
+import org.openehr.bmm.core.BmmEnumerationInteger;
+import org.openehr.bmm.core.BmmEnumerationString;
+import org.openehr.bmm.core.BmmModel;
+import org.openehr.bmm.core.BmmProperty;
+import org.openehr.bmm.persistence.validation.BmmDefinitions;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class MetaModel implements MetaModelInterface {
+
+    private ModelInfoLookup selectedModel;
+    private BmmModel selectedBmmModel;
+    private AomProfile selectedAomProfile;
+
+    public MetaModel(ModelInfoLookup selectedModel, BmmModel selectedBmmModel) {
+        this(selectedModel, selectedBmmModel, null);
+    }
+
+    public MetaModel(ModelInfoLookup selectedModel, BmmModel selectedBmmModel, AomProfile selectedAomProfile) {
+        this.selectedModel = selectedModel;
+        this.selectedBmmModel = selectedBmmModel;
```

Comment:
Either selectedModel or selectedBmmModel can be null. If both are null, all methods silently return false 
by default. This is probably not desired behaviour.

Since there are no methods to set these models later, perhaps throw an error in the constructor that one of them has to be set? Alternatively throw an error in each of these methods, but it's easier to handle errors when actually creating the MetaModel